### PR TITLE
`mock_call` appendix revamp

### PR DESCRIPTION
**Pull request by Arcticae**
References #1439 

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: snforge_std/src/cheatcodes.cairo [hunk 1]
```diff
@@ -110,6 +110,17 @@ fn stop_elect(target: CheatTarget) {
...
 
+
+/// Mocks contract call to a `function_selector` of a contract at the given address, for `n_times` first calls that are made
+/// to the contract.
+/// A call to function `function_selector` will return data provided in `ret_data` argument.
+/// An address with no contract can be mocked as well.
+/// An entrypoint that is not present on the deployed contract is also possible to mock.
+/// Note that the function is not meant for mocking internal calls - it works only for contract entry points.
+/// - `contract_address` - target contract address
+/// - `function_selector` - selector of the function in a contract at the `contract_address` that will be mocked
```
Comment (drknzz) [inline snforge_std/src/cheatcodes.cairo:121]: Should it include anything about possible obtaining it from the `selector!` macro?

File: snforge_std/src/cheatcodes.cairo [hunk 2]
```diff
@@ -144,6 +162,9 @@ fn start_mock_call<T, impl TSerde: core::serde::Serde<T>, impl TDestruct: Destru
     handle_cheatcode(cheatcode::<'mock_call'>(inputs.span()));
 }
 
+/// Cancels the `mock_call` / `start_mock_call` for the function `function_selector` of a contract at the given address.
+/// - `contract_address` - target contract address
+/// - `function_selector` - selector of the function
```
Comment (drknzz) [inline snforge_std/src/cheatcodes.cairo:167]: image

File: docs/src/SUMMARY.md [hunk 3]
```diff
@@ -46,6 +46,7 @@
 * [Cheatcodes Reference](appendix/cheatcodes.md)
     * [CheatTarget](appendix/cheatcodes/cheat_target.md)
     * [CheatSpan](appendix/cheatcodes/cheat_span.md)
+    * [mock](appendix/cheatcodes/mock_call.md)
```
Comment (drknzz) [inline docs/src/SUMMARY.md:49]:
```suggestion
    * [mock_call](appendix/cheatcodes/mock_call.md)
```
probably should be then

File: snforge_std/src/cheatcodes.cairo (hunk 1 above)
Comment (Arcticae) [reply snforge_std/src/cheatcodes.cairo:121]: yeah i think i'll include that

File: snforge_std/src/cheatcodes.cairo (hunk 2 above)
Comment (Arcticae) [reply snforge_std/src/cheatcodes.cairo]: love that meme

**Review by drknzz — APPROVED** [review]

**Pull request merged**